src/Translator.php

```diff
@@ -143,11 +143,12 @@ public function getGlossaryLanguages(): array
 
     /**
      * Translates specified text string or array of text strings into the target language.
-     * @param $texts string|string[] A single string or array of strings containing input texts to translate.
+     * @param string|string[] $texts A single string or array of strings containing input texts to translate.
      * @param string|null $sourceLang Language code of input text language, or null to use auto-detection.
      * @param string $targetLang Language code of language to translate into.
      * @param array $options Translation options to apply. See \DeepL\TranslateTextOptions.
      * @return TextResult|TextResult[] A TextResult or array of TextResult objects containing translated texts.
+     * @phpstan-return ($texts is array ? TextResult[] : TextResult)
```

Comment:
Is there a reason to prefer adding a new line with `@phpstan-return` over changing l. 150 to 

```php
* @return ($texts is array ? TextResult[] : TextResult) A TextResult or array of TextResult objects containing translated texts.
```

? This works with psalm and phpstan (I assume your version does too) - psalm's official docs say to use and additional `psalm-return` line and phpstan to use an additional `phpstan-return` line. Do both also work with PHPStorm's type checker?

Reply:
Maybe modifying this line will interfere with less sophisticated tools e.g. IDEs?

Reply:
> Maybe modifying this line will interfere with some less sophisticated tools e.g. IDEs?

Yes, I prefer to let `@return` for other tool/IDE less sophisticated and a a `@phpstan-return` for sophisticated tools like psalm/phpstan.